File: algorithms/MergeTwoSortedLists/MergeTwoSortedLists.c
struct ListNode* mergeTwoLists(struct ListNode* l1, struct ListNode* l2) {
    struct ListNode *cur;
    struct ListNode *res = cur;
    if(!l1 && l2)
            return l2;
        if(l1 && !l2)
            return l1;
        if(!l1 && !l2)
            return NULL;
    while (l1 && l2) {
        if(l1->val <= l2->val)
        {
            cur->next = l1;
            l1 = l1->next;
        }
        else
        {
            cur->next = l2;
            l2 = l2->next;
        }
        cur = cur->next;
    }
    if(l1)
        cur->next = l1;
    if(l2)
        cur->next = l2;
  
    return res->next;
}

File: algorithms/MergeTwoSortedLists/MergeTwoSortedLists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param {ListNode} l1
    # @param {ListNode} l2
    # @return {ListNode}
    def mergeTwoLists(self, l1, l2):
    	res = cur = ListNode(0)
    	while li and l2:
    		if l1.val < l2.val:
    			cur.next = l1
    			l1 = l1.next
    		else:
    			cur.next = l2
    			l2 = l2.next
    		cur = cur.next
    	cur.next = l1 or l2
    	return res.next